addons/pos_self_order/models/pos_config.py

```diff
@@ -28,6 +29,12 @@ def _self_order_default_image(self) -> bytes:
     def _self_order_kiosk_default_languages(self):
         return self.env["res.lang"].get_installed()
 
+    def _self_order_default_user(self):
+        user_ids = self.env["res.users"].search([])
```

Comment:
I debugged the issue from the runbot. I thought it's because no default user is probably assigned to the config but that's wrong. The issue was because the default user calculated here can be from different company.

We should search on the users that belong to the same company as the config.